sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_search_sessions_v1/query.sql

```diff
@@ -0,0 +1,215 @@
+WITH events_unnested AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    sample_id,
+    client_info.client_id,
+    timestamp AS event_timestamp,
+    normalized_channel,
+    normalized_country_code,
+    name AS event_name,
+    CASE
+      WHEN name = 'engagement'
+        AND (
+          (
+            mozfun.map.get_key(extra, "engagement_type") IN (
+              "click",
+              "drop_go",
+              "enter",
+              "go_button",
+              "paste_go"
+            )
+          )
+        )
+        THEN 'engaged'
+      WHEN name = 'abandonment'
+        THEN 'abandoned'
+      WHEN name = 'engagement'
+        AND (
+          (
+            mozfun.map.get_key(extra, "engagement_type") NOT IN (
+              "click",
+              "drop_go",
+              "enter",
+              "go_button",
+              "paste_go"
+            )
+          )
+        )
+        THEN "annoyance"
+      ELSE NULL
+    END AS search_session_type,
+    SPLIT(mozfun.map.get_key(extra, "results"), ',')[OFFSET(0)] AS result_type,
+    SPLIT(mozfun.map.get_key(extra, "results"), ',') AS results,
+    mozfun.map.get_key(extra, "selected_result") AS selected_results,
```

Comment:
```suggestion
    SPLIT(mozfun.map.get_key(extra, "results"), ',')[OFFSET(0)] AS result_type,
    mozfun.norm.result_type_to_product_name(SPLIT(mozfun.map.get_key(extra, "results"), ',')[OFFSET(0)]) AS product_result_type,
    SPLIT(mozfun.map.get_key(extra, "results"), ',') AS results,
    ARRAY(SELECT mozfun.norm.result_type_to_product_name(x) FROM UNNEST(SPLIT(mozfun.map.get_key(extra, "results"), ',')) AS x) AS product_results,
    mozfun.map.get_key(extra, "selected_result") AS selected_results,
    mozfun.norm.result_type_to_product_name(mozfun.map.get_key(extra, "selected_result")) AS product_selected_results,
```
Using the UDF, so the mapping logic is in one consistent place.

Reply:
This is a minor nit, but I suggest removing the plural from `selected_results`/`product_selected_results`. These are not column names in the final table, but I think it would help readability of the query - these fields contain a single result.